femtools/Field_derivatives.F90

```diff
@@ -42,16 +42,280 @@ module field_derivatives
       module procedure grad_scalar, grad_vector, grad_vector_tensor
     end interface grad
 
+    interface dg_ele_grad
+      module procedure dg_ele_grad_scalar, dg_ele_grad_vector
+    end interface dg_ele_grad
+
+    interface dg_ele_grad_at_quad
+      module procedure dg_ele_grad_at_quad_scalar, dg_ele_grad_at_quad_vector
+    end interface dg_ele_grad_at_quad
+
     private
 
     public :: strain_rate, differentiate_field, grad, compute_hessian, &
       domain_is_2d, patch_type, get_patch_ele, get_patch_node, get_quadratic_fit_qf, curl, &
-      get_quadratic_fit_eqf, div, u_dot_nabla, get_cubic_fit_cf, differentiate_field_lumped
+      get_quadratic_fit_eqf, div, u_dot_nabla, get_cubic_fit_cf, differentiate_field_lumped, &
+      dg_ele_grad_at_quad, dg_ele_grad
 
     public :: compute_hessian_qf, compute_hessian_eqf, compute_hessian_var
     
     contains
 
+    function dg_ele_grad_scalar(field, ele, X, bc_value, bc_type) result (loc_grad)
+      ! Return the grad of field at the quadrature points of
+      ! ele_number. dn is the transformed element gradient. 
+      ! including interface terms for dg discretisations based upon
+      ! N_i N_j grad_u = N_i delta u_h - 
+      !                  ({N_i} (u_h^-n^- + u_h^+n^+)) on internal faces -
```

Comment:
In the code below it seems to have a factor of 0.5 for internal facets. Why is that?